### PR TITLE
Signature verification requests routing

### PR DESCRIPTION
**Pull request by nplasterer**
"Create `MultiChainSignatureVerifier` that routes signature verification requests to `RpcSmartContractWalletVerifier` instances configured with a RPC URL for the correct chain."

Part of https://github.com/xmtp/libxmtp/issues/1070

Depends on https://github.com/xmtp/proto/pull/211

### CODE REVIEW

File: xmtp_id/src/lib.rs
Comment (neekolas) [inline xmtp_id/src/lib.rs:1]: Why did we change the permissions on this file?

File: xmtp_mls/src/builder.rs [hunk 1]
```diff
@@ -115,9 +115,8 @@ where
         }
 
         let scw_verifier = self.scw_verifier.take().unwrap_or_else(|| {
-            // TODO:nm Enforce that everyone provides this
-            Box::new(RpcSmartContractWalletVerifier::new(
-                "https://fixme.com".to_string(),
+            Box::new(ChainSmartContractWalletVerifier::new_from_file(
```
Comment (neekolas) [inline xmtp_mls/src/builder.rs:118]: Nice

**Comment by insipx** [comment]
Before merging should:
- clean up the `Box<dyn SmartContractVerifierTrait>` (Clone impl, where on client its stored, etc.)
- write a test maybe

**Comment by codabrink** [comment]
Depends on https://github.com/xmtp/proto/pull/211

**Review by insipx — APPROVED** [review]
looks good except for the try_join!

File: mls_validation_service/src/handlers.rs [hunk 2]
```diff
@@ -159,6 +198,50 @@ async fn validate_inbox_id_key_package(
...
+                .await
+                .map_err(|e| GrpcServerError::Signature(SignatureError::VerifierError(e)))?;
+
+            Ok(valid)
+        };
+
+        responses.push(handle);
+    }
+
+    let responses: Vec<_> = try_join_all(responses)
```
[the remark this answers is not in the document]
Comment (insipx) [reply mls_validation_service/src/handlers.rs:234]: Could also return a Vec<> of errors or something. Might be ok to do it in another PR as long as an issue is created

[the remark this answers is not in the document]
Comment (codabrink) [reply mls_validation_service/src/handlers.rs:234]: Giving that a little more thought, I should add an error message field to the validation response, so this doesn't turn into a black box. Let me get a quick proto pr up.

[the remark this answers is not in the document]
Comment (codabrink) [reply mls_validation_service/src/handlers.rs]: https://github.com/xmtp/proto/pull/214

**Pull request merged**